### PR TITLE
Hide values of environment variables

### PR DESCRIPTION
**Pull request by saraedum**
these sometimes contain secrets that leak when conda-build is used in a CI environment, see https://github.com/conda-forge/conda-smithy/issues/1162.

### CODE REVIEW

**Comment by saraedum** [comment]
Should I add a news entry for a small change like this?

**Comment by jjhelmus** [comment]
Can this be suppression be placed behind a configuration option or flag?  I find the listing of environment variables contents very useful when debugging and would prefer to keep them but understand the need to suppress them in cases when leaking secrets is a concern.

**Comment by jjhelmus** [comment]
An alternative is add `export PYTHONWARNINGS=ignore` before calling `conda build`.  This will suppress printing of all warning messages.

**Comment by saraedum** [comment]
@jjhelmus: Thanks for the feedback. What do you think about something like this? I could go all the way and allow `script_env` to optionally be a dictionary `{ name: … , hidden: true }` but I am not sure if that would be overengineering it a bit here?

**Comment by saraedum** [comment]
> An alternative is add `export PYTHONWARNINGS=ignore` before calling `conda build`. This will suppress printing of all warning messages.

I guess it's the same argument here that this might hide some warnings that people would like to see. But sure, it's a workaround.

**Comment by marcelotrevisani** [comment]
Maybe an option which will be passed by the command line would be good. What do you think?
Something like
```
conda build --suppress-variables
```

**Comment by saraedum** [comment]
If you guys don't mind that I add another command line switch that also works of course.

**Comment by saraedum** [comment]
@marcelotrevisani @jjhelmus Does this look alright to you now?

**Comment by marcelotrevisani** [comment]
> @marcelotrevisani @jjhelmus Does this look alright to you now?

I think that would be a nice addition. However, could you please fix those failing tests? They are related to the ``--suppress-variables``

**Comment by saraedum** [comment]
Note that `build.sh` needs to be around for the warning to print. Not sure why.

**Comment by saraedum** [comment]
@marcelotrevisani would you mind having another look now?

File: tests/test_api_build.py [hunk 1]
```diff
@@ -1408,3 +1408,26 @@ def test_ignore_verify_codes(testing_config):
     # this recipe intentionally has a license error.  If ignore_verify_codes works,
     #    it will build OK.  If not, it will error out.
     api.build(recipe_dir, config=testing_config)
+
+
+def test_script_env_warnings(testing_config, recwarn):
```
Comment (marcelotrevisani) [inline tests/test_api_build.py]: Thanks for adding this test! I really appreciate that! :)

File: tests/test_api_build.py [hunk 2]
```diff
@@ -1408,3 +1408,26 @@ def test_ignore_verify_codes(testing_config):
...
+    def assert_keyword(keyword):
+        messages = [str(w.message) for w in recwarn.list]
+        assert any([token in m and keyword in m for m in messages])
+        recwarn.clear()
+
+    api.build(recipe_dir, config=testing_config)
+    assert_keyword('undefined')
+
+    os.environ[token] ='SECRET'
+    try:
```
Comment (marcelotrevisani) [inline tests/test_api_build.py]: I just think that the ``try`` here is not necessary, is it?

Comment (saraedum) [reply tests/test_api_build.py]: I don't know pytest well enough. Just thought I'd clean up after myself.

Comment (saraedum) [reply tests/test_api_build.py]: So, how does this work? I don't need to cleanup or I should just drop the `try - finally` part of it?

Comment (marcelotrevisani) [reply tests/test_api_build.py]: I think that is fine then, as it is used just here should be ok
However, you can create a fixture and do the setup and teardown there. But as you are just adding and removing an env variable that is ok
:p

**Comment by marcelotrevisani** [comment]
For me that is fine! As @jjhelmus was reviewing it as well, I would like to see if he also agree with the modifications

**Comment by msarahan** [comment]
Thanks @saraedum @jjhelmus @marcelotrevisani for the hard work on this.

**Pull request merged**